test/store-test.js

```diff
@@ -62,24 +62,39 @@ describe('hoodie.store', function () {
       expect(hasChanges).to.equal(true)
     })
 
+    // sanity check
+    .execute(function dafug () {
+      return hoodie.account.username
+    }).then(toValue)
+    .then(function (username) {
+      expect(username).to.equal(null)
+    })
+
     .executeAsync(function (done) {
       hoodie.account.signUp({
         username: 'storetest',
         password: 'secret'
       })
 
+      .then(function () {
+        return hoodie.account.signIn({
+          username: 'storetest',
+          password: 'secret'
+        })
+      })
+
       .then(done, done)
     })
 
     .waitUntil(function () {
       return this.execute(function storeHasNoLocalChanges () {
         return hoodie.store.hasLocalChanges() === false
-      })
+      }).then(toValue)
```

Comment:
this `waitUntil` actually always passed, because `this.execute` always returnes an object, so the `toValue` is critical. I must have not made sure that tests failed before they passed, sorry